scapy/pton_ntop.py

```diff
@@ -80,10 +80,10 @@ def _inet6_pton(addr):
 def inet_pton(af, addr):
     """Convert an IP address from text representation into binary form."""
     # Will replace Net/Net6 objects
-    if not isinstance(addr, str):
+    addr = plain_str(addr)
+    if not isinstance(addr, six.string_types):
```

Comment:
I don't think this test is useful (this line and the next one could be removed). `plain_str()` should always return a `str`. Or did I miss something?

Reply:
Currents behavior is to return the argument when not a bytes. Net6/Net is not a byte

Reply:
OK. Maybe we could change the behaviour of `plain_str()`? I'm not sure...

Reply:
`plain_str` is called many times when loading scapy, I don't think we want to overload it with specific cases such as this one... Even though it does not handle all cases, it's very fast for now.

Reply:
I'm not saying we should overload it, but I'd assume `plain_str()` to always return a `str` object!

Reply:
Updated